dotcom-rendering/src/lib/useOnce.ts

```diff
@@ -1,11 +1,16 @@
 import { useEffect, useState } from 'react';
 
+type ArrayOfAtLeastOne<T> = [T, ...T[]];
+
 /**
  * Ensures that the given task is only run once and only after all items in waitFor are defined
  * @param {Function} task - The task to execute once
  * @param {Array} waitFor - An array of variables that must be defined before the task is executed
  * */
-export const useOnce = (task: () => void, waitFor: unknown[]): void => {
+export const useOnce = (
+	task: () => void,
+	waitFor: ArrayOfAtLeastOne<unknown>,
```

Comment:
Can we just inline to `[T, ...T[]]`?

Otherwise, if you want to create a helper, you could [add it to `tuples.ts`](https://github.com/guardian/dotcom-rendering/blob/03cb41384aa5285dd5b4fc18ae83e23cd6266b55/dotcom-rendering/src/lib/tuple.ts#L2).

Reply:
I think naming might help others understand what this does. But will move to `tuple.ts` as a better option.